### PR TITLE
Update access levels for more classes

### PR DESCRIPTION
**Pull request by timkimadobe**
## Description

This PR updates: 
1. Access levels to various classes and methods 
2. Transfers the AnyCodable JSON comparison system unit test classes from Edge extension to the test utils repo (as-is)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Sources/NetworkRequestHelper.swift [hunk 1]
```diff
@@ -209,7 +209,7 @@ class NetworkRequestHelper {
     }
 }
 
-extension NetworkRequest {
+public extension NetworkRequest {
```
Comment (cdhoffmann) [inline Sources/NetworkRequestHelper.swift:212]: Should we update the access levels for the methods as well? Applies to the other classes in this PR

**Review by timkimadobe — COMMENTED** [review]
Thanks for the review @cdhoffmann! Addressed feedback in latest update

File: Sources/RealNetworkService.swift [hunk 2]
```diff
@@ -17,6 +17,10 @@ import XCTest
 /// Overriding NetworkService used for tests that require real outgoing network requests
 public class RealNetworkService: NetworkService {
     private let helper: NetworkRequestHelper = NetworkRequestHelper()
+    
+    public override init() {
+        super.init()
```
Comment (emdobrin) [inline Sources/RealNetworkService.swift:22]: what is the default initialization required here?

**Review by timkimadobe — COMMENTED** [review]
Thanks for the review @emdobrin! Updated based on feedback

**Review by emdobrin — APPROVED** [review]

**Pull request merged**